### PR TITLE
docs(markdown): migrate docs from rst to md

### PR DESCRIPTION
**Pull request by hetangmodi-crest**
Migrated docs from `readthedocs.io` to `github.io`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -49,8 +49,6 @@ pytest11 = { "ucc-smartx" = "pytest_splunk_addon_ui_smartx.plugin" }
 [tool.poetry.group.dev.dependencies]
 pytest = ">=5.4, <7.3"
 Sphinx = "*"
```
Comment (artemrys) [inline pyproject.toml:51]: We can get rid of `Sphinx` dependency as well, right?

**Review by artemrys — CHANGES_REQUESTED** [review]
Let's clean up `README.md` file as well - let's have it similar to https://github.com/splunk/pytest-splunk-addon/blob/main/README.md.

If instructions listed there are valid - let's move them in the documentation, if not - let's remove them.

File: pyproject.toml (hunk 1 above)
Comment (hetangmodi-crest) [reply pyproject.toml]: Removed the `Sphinx` library, and updated ucc-gen library.

**Comment by hetangmodi-crest** [comment]
> Let's clean up `README.md` file as well - let's have it similar to https://github.com/splunk/pytest-splunk-addon/blob/main/README.md.
> 
> If instructions listed there are valid - let's move them in the documentation, if not - let's remove them.

The content was related to the design of the framework, which I think isn't relevant to developers using this framework, and hence has been removed.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          python-version: 3.8
+      - run: |
+          curl -sSL https://install.python-poetry.org | python3 - --version 1.5.1
+          pip install mkdocs mkdocs-material mkdocstrings-python
+      - name: Deploy to GitHub Pages
+        if: github.ref_name == 'main'
+        run: mkdocs gh-deploy --force --strict
+      - name: Build Docs
+        if: github.ref_name != 'main'
+        run: mkdocs build
```
Comment (artemrys) [inline .github/workflows/docs.yml:29]: Add `--strict` here as well please. And please make sure it build locally with `--strict` as well.

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+```{toctree}
+:caption: 'Contents:'
+:maxdepth: 3
+
+overview
+how_to_use
+getting_started
+api_reference
+```
```
Comment (artemrys) [inline docs/index.md:16]: It does not seem right. `overview.md` looks like a nice page to default to when you open a SmartX documentation.

File: docs/overview.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# Overview
+
+UCC based Add-on UI automation framework
+Confluence Page link:
+[SmartX Repo](https://confluence.splunk.com/display/PROD/SmartX+UI+Automation+Framework+for+Ucc+based+Add-ons)
```
Comment (artemrys) [inline docs/overview.md:5]: The link does not work anymore, please update it.

File: docs/overview.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+The Framework could be cloned from here: [SmartX Repo](https://github.com/splunk/addon-factory-smartx-ui-test-library)
+
+```console
+1. git clone https://github.com/splunk/addon-factory-smartx-ui-test-library.git
+2. python3 -m venv .venv
+3. source .venv/bin/activate
+4. cd addon-factory-smartx-ui-test-library
+5. poetry install
+```
```
Comment (artemrys) [inline docs/overview.md:65]: Let's mention that people can download SmartX using PyPI. I don't even think those instructions are even correct.

File: docs/overview.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+You could also download SmartX from PyPi as well: [SmartX PyPI](https://pypi.org/project/pytest-splunk-addon-ui-smartx/)
```
Comment (artemrys) [inline docs/overview.md:67]:
```suggestion
You could also download SmartX from PyPI as well: [SmartX PyPI](https://pypi.org/project/pytest-splunk-addon-ui-smartx/)
```

File: mkdocs.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+site_name: SmartX
+site_author: Splunk
+site_url: "https://github.com/splunk/addon-factory-smartx-ui-test-library/"
```
Comment (artemrys) [inline mkdocs.yml:18]:
```suggestion
site_url: "https://splunk.github.io/addon-factory-smartx-ui-test-library/"
```

File: mkdocs.yml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+nav:
+  - Home: "index.md"
+  - Overview: "overview.md"
+  - How To Use: "how_to_use.md"
+  - Getting Started:
+    - "getting_started/index.md"
+    - Setup: "getting_started/setup.md"
+    - Test Page Creation: "getting_started/test_page.md"
+    - Test Cases Creation: "getting_started/test_case.md"
+  - API Reference:
```
Comment (artemrys) [inline mkdocs.yml:69]: What about all the `.md` pages we have?

When I build the docs locally I see the following message:

```
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
             - api_reference/components/base_component.md
             - api_reference/components/conf_table.md
             - api_reference/components/dropdown.md
             - api_reference/components/entity.md
             - api_reference/components/input_table.md
             - api_reference/components/login.md
             - api_reference/components/table.md
             - api_reference/components/tabs.md
             - api_reference/components/controls/base_control.md
             - api_reference/components/controls/button.md
             - api_reference/components/controls/checkbox.md
             - api_reference/components/controls/learn_more.md
             - api_reference/components/controls/message.md
             - api_reference/components/controls/multi_select.md
             - api_reference/components/controls/oauth_select.md
             - api_reference/components/controls/single_select.md
             - api_reference/components/controls/textbox.md
             - api_reference/components/controls/toggle.md
             - api_reference/pages/logging.md
             - api_reference/pages/login.md
             - api_reference/pages/page.md
             - api_reference/pages/proxy.md
```

File: docs/api_reference/components/entity.md [hunk 9]
```diff
@@ -1,6 +1,3 @@
-Entity
-======
+# Entity
```
Comment (artemrys) [inline docs/api_reference/components/entity.md:1]: Do we need `automodule` here?

File: docs/api_reference/index.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
+
+The following is the API Reference for SmartX:
```
Comment (artemrys) [inline docs/api_reference/index.md:2]: Would it be a bit more discoverable if we have this structure in the navigation (`mkdocs.yml>nav`) instead?

Comment (artemrys) [reply docs/api_reference/index.md:2]: image

**Review by artemrys — CHANGES_REQUESTED** [review]

File: docs/api_reference/components/entity.md (hunk 9 above)
Comment (hetangmodi-crest) [reply docs/api_reference/components/entity.md:1]: Missed that file to update. It is now updated.

File: docs/api_reference/index.md (hunk 10 above)
Comment (hetangmodi-crest) [reply docs/api_reference/index.md:2]: I had mimicked the current doc of this framework, that it expands when clicked, but I think we don't have it for `.md` files. I have updated the nav to contain all.

File: docs/index.md (hunk 3 above)
Comment (hetangmodi-crest) [reply docs/index.md]: Updated the file.

**Review by artemrys — APPROVED** [review]

**Pull request merged**